code/modules/ninja/suit/suit.dm

```diff
@@ -7,80 +7,184 @@ Contents:
+
+		//Important parts of the suit.
+	var/mob/living/carbon/human/affecting = null
+	var/obj/item/stock_parts/cell/cell
+	var/datum/effect_system/spark_spread/spark_system
+	var/list/stored_research = list()//For stealing station research.
+	var/obj/item/disk/tech_disk/t_disk//To copy design onto disk.
+	var/obj/item/energy_katana/energyKatana //For teleporting the katana back to the ninja (It's an ability)
+
+		//Other articles of ninja gear worn together, used to easily reference them after initializing.
+	var/obj/item/clothing/head/helmet/space/space_ninja/n_hood
+	var/obj/item/clothing/shoes/space_ninja/n_shoes
+	var/obj/item/clothing/gloves/space_ninja/n_gloves
+
+		//Main function variables.
+	var/s_initialized = 0 //Suit starts off.
+	var/s_coold = 0 //If the suit is on cooldown. Can be used to attach different cooldowns to abilities. Ticks down every second based on suit ntick().
+	var/s_cost = 5 //Base energy cost each ntick.
+	var/s_acost = 25 //Additional cost for additional powers active.
+	var/s_delay = 40 //How fast the suit does certain things, lower is faster. Can be overridden in specific procs. Also determines adverse probability.
+	var/a_transfer = 20 //How much radium is used per adrenaline boost.
+	var/a_maxamount = 7 //Maximum number of adrenaline boosts.
+	var/s_maxamount = 20 //Maximum number of smoke bombs.
+
+		//Support function variables.
+	var/stealth = FALSE//Stealth off.
+	var/s_busy = FALSE//Is the suit busy with a process? Like AI hacking. Used for safety functions.
+
+		//Ability function variables.
+	var/s_bombs = 10 //Number of smoke bombs.
+	var/a_boost = 3 //Number of adrenaline boosters.
+
+
+/obj/item/clothing/suit/space/space_ninja/New()
+	..()
+
+	//Spark Init
+	spark_system = new
+	spark_system.set_up(5, 0, src)
+	spark_system.attach(src)
+
+	//Research Init
+	stored_research = new()
+	for(var/T in subtypesof(/datum/tech))//Store up on research.
+		stored_research += new T(src)
+
+	//Cell Init
+	cell = new/obj/item/stock_parts/cell/high
+	cell.charge = 9000
+	cell.name = "black power cell"
+	cell.icon_state = "bscell"
+
+//Simply deletes all the attachments and self, killing all related procs.
+/obj/item/clothing/suit/space/space_ninja/proc/terminate()
+	qdel(n_hood)
+	qdel(n_gloves)
+	qdel(n_shoes)
+	qdel(src)
+
+
+//Randomizes suit parameters.
+/obj/item/clothing/suit/space/space_ninja/proc/randomize_param()
+	s_cost = rand(1, 20)
+	s_acost = rand(20, 100)
+	s_delay = rand(10, 100)
+	s_bombs = rand(5, 20)
+	a_boost = rand(1, 7)
+
+
+//This proc prevents the suit from being taken off.
+/obj/item/clothing/suit/space/space_ninja/proc/lock_suit(mob/living/carbon/human/H)
+	if(!istype(H))
+		return FALSE
+	if(H.mind.special_role != "Space Ninja")
```

Comment:
This seems a bit irritating to have hardcoded into the suit, for admin purposes. Might wanna make this check able to be skipped with a var. (I realize that they get the player panel "make ninja" but they may not want to generate a full ninja, just play with the suit)